index-handlers/src/main/java/no/sikt/nva/nvi/index/SearchNviCandidatesHandler.java

```diff
@@ -49,20 +68,45 @@ protected PaginatedSearchResult<NviCandidateIndexDocument> processInput(Void inp
         var offset = extractQueryParamOffsetOrDefault(requestInfo);
         var size = extractQueryParamSizeOrDefault(requestInfo);
         var filter = extractQueryParamFilterOrDefault(requestInfo);
-        var affiliations = extractQueryParamAffiliationsOrDefault(requestInfo);
         var excludeSubUnits = extractQueryParamExcludeSubUnitsOrDefault(requestInfo);
-        var customer = requestInfo.getTopLevelOrgCristinId().orElseThrow();
+        var topLevelOrg = requestInfo.getTopLevelOrgCristinId().orElseThrow();
+        var affiliations = Optional.ofNullable(extractQueryParamAffiliations(requestInfo))
+                               .orElse(List.of(topLevelOrg));
         var username = requestInfo.getUserName();
         var year = extractQueryParamPublicationDateOrDefault(requestInfo);
 
+        assertUserIsAllowedToSearchAffiliations(affiliations, topLevelOrg);
+
         var candidateSearchParameters = new CandidateSearchParameters(affiliations, excludeSubUnits, filter, username,
-                                                                      year, customer, offset, size);
+                                                                      year, topLevelOrg, offset, size);
 
         return attempt(() -> openSearchClient.search(candidateSearchParameters))
```

Comment:
I think we will soon need a method that extracts all searchparams from requestInfo, something like:

`CandidateSearchParams.fromRequestInfo(requestInfo)`